### PR TITLE
SMS Sending Capability

### PR DESCRIPTION
**Pull request by daltonfury42**
https://github.com/SimplQ/simplQ-frontend/issues/257

For now, a simple one screen app that can send sms:

Screenshot 2020-12-26 at 5 44 23 PM

https://stackoverflow.com/questions/6869961/how-to-send-and-receive-sms-from-android-app
https://github.com/android/permissions-samples/tree/main/RuntimePermissionsBasic

### CODE REVIEW

File: app/src/main/java/me/simplq/MainActivity.java [hunk 1]
```diff
@@ -1,14 +1,111 @@
...
+    EditText txtMessage;
+    private View mLayout;
+    private static final int PERMISSION_REQUEST_SMS = 0;
+
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        mLayout = findViewById(android.R.id.content);;
```
Comment (thehamzarocks) [inline app/src/main/java/me/simplq/MainActivity.java]: I suppose we're not using any formatter 😅

File: app/src/main/java/me/simplq/MainActivity.java [hunk 2]
```diff
@@ -1,14 +1,111 @@
...
+                        Toast.LENGTH_SHORT).show();
+        });
+
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.SEND_SMS)
+                == PackageManager.PERMISSION_GRANTED) {
+            // Permission is already available, start camera preview
+            // TODO: Show status on the app
+        } else {
+            // Permission is missing and must be requested.
+            requestCameraPermission();
```
Comment (thehamzarocks) [inline app/src/main/java/me/simplq/MainActivity.java]: Camera permission?

File: app/src/main/java/me/simplq/MainActivity.java [hunk 3]
```diff
@@ -1,14 +1,111 @@
...
+        PendingIntent pi = PendingIntent.getActivity(this, 0,
+                new Intent(this, MainActivity.class), 0);
+        SmsManager sms = SmsManager.getDefault();
+        // TODO log/notify on failure
+        sms.sendTextMessage(phoneNumber, null, message, pi, null);
+    }
+
+    private void requestCameraPermission() {
+        // Permission has not been granted and must be requested.
+        if (ActivityCompat.shouldShowRequestPermissionRationale(this,
```
Comment (thehamzarocks) [inline app/src/main/java/me/simplq/MainActivity.java:70]: Why camera though>

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1,3 +1,7 @@
 <resources>
     <string name="app_name">SimplQ</string>
+    <string name="ok">OK</string>
+    <string name="sms_access_required">Permission to send SMS is required to send SMS to people who join the queue</string>
+    <string name="sms_unavailable">This app won\'t work without granding SMS permissions.</string>
```
Comment (thehamzarocks) [inline app/src/main/res/values/strings.xml]: granting?

File: app/src/main/java/me/simplq/MainActivity.java (hunk 1 above)
Comment (daltonfury42) [reply app/src/main/java/me/simplq/MainActivity.java]: Have to set it up...

File: app/src/main/java/me/simplq/MainActivity.java (hunk 2 above)
Comment (daltonfury42) [reply app/src/main/java/me/simplq/MainActivity.java]: Copy paste effect, I've corrected.

File: app/src/main/java/me/simplq/MainActivity.java (hunk 3 above)
Comment (daltonfury42) [reply app/src/main/java/me/simplq/MainActivity.java:70]: Ctrl+CV

**Pull request merged**